app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -382,46 +413,23 @@ class BrowserFragment : Fragment(), BackHandler {
                     activity,
                     view.nestedScrollQuickAction,
                     view.swipeRefresh,
-                    onItemTapped = { actionEmitter.onNext(SearchAction.ToolbarMenuItemTapped(it)) }
+                    onItemTapped = { browserInteractor.onBrowserToolbarMenuItemTapped(it) }
                 ),
                 owner = this,
                 view = view)
         }
 
-        toolbarComponent.setOnSiteSecurityClickedListener {
+        browserToolbarView.view.setOnSiteSecurityClickedListener {
             showQuickSettingsDialog()
         }
 
-        val appLink = requireComponents.useCases.appLinksUseCases.appLinkRedirect
-        quickActionSheetStore = StoreProvider.get(this) {
-            QuickActionSheetStore(
-                QuickActionSheetState(
-                    readable = getSessionById()?.readerable ?: false,
-                    bookmarked = findBookmarkedURL(getSessionById()),
-                    readerActive = getSessionById()?.readerMode ?: false,
-                    bounceNeeded = false,
-                    isAppLink = getSessionById()?.let { appLink.invoke(it.url).hasExternalApp() } ?: false
-                )
-            )
-        }
-
-        val quickActionSheetView = QuickActionView(
-            view.nestedScrollQuickAction,
-
-            QuickActionInteractor(
-                context!!,
-                DefaultReaderModeController(readerViewFeature),
-                quickActionSheetStore,
-                shareUrl = ::shareUrl,
-                bookmarkTapped = {
-                    lifecycleScope.launch { bookmarkTapped(it) }
-                },
-                appLinksUseCases = requireComponents.useCases.appLinksUseCases
-            )
-        )
-
-        consumeFrom(quickActionSheetStore) {
-            quickActionSheetView.update(it)
+        browserStore.observe(view) {
```

Comment:
`consumeFrom` was right here so we can replace this now with:
```kotlin
consumeFrom(browserStore) {
  quickActionSheetView.update(it)
  browserToolbarView.update(it)
}
```

See also: https://github.com/mozilla-mobile/fenix/pull/4294

Reply:
Thanks for adding test! Let's just address ^^ and squash/land this. Did another round of testing too!

Reply:
Resolved this 😄